src/Shared/src/ActivatorUtilities/ActivatorUtilities.cs

```diff
@@ -403,16 +403,20 @@ public object CreateInstance(IServiceProvider provider)
                     }
                 }
 
+#if NETCOREAPP3_0
+                return _constructor.Invoke(BindingFlags.DoNotWrapExceptions, binder: null, parameters: _parameterValues, culture: null);
+#else
                 try
                 {
                     return _constructor.Invoke(_parameterValues);
                 }
-                catch (TargetInvocationException ex)
+                catch (TargetInvocationException ex) when (ex.InnerException != null)
```

Comment:
Is TargetInvocationException.InnerException ever in fact null?

Reply:
No. I don't think so.